Tests/Tests/Core/Extensions/Other/XCTestCaseTests.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  XCTestCaseTests.swift
+//  PovioKit_Tests
+//
+//  Created by Borut Tomazin on 15/09/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import XCTest
+
+final class XCTestCaseTests: XCTestCase {
+  func test_httpBody_failsOnDifferentDataSets() {
+    let data1 = "TestData".data(using: .utf8)
+    let data2 = "Povio🚀".data(using: .utf8)
+    
+    var firstUrlRequest = anyRequest()
+    firstUrlRequest.httpBody = data1
+    
+    var secondUrlRequest = anyRequest()
+    secondUrlRequest.httpBody = data2
+    
+    // Test passes even though httpBody is different.
+    XCTAssertEqual(firstUrlRequest, secondUrlRequest)
+    
+    XCTExpectFailure("Expected test to fail since httpBody is different.") {
+      XCTAssertEqual(firstUrlRequest.httpBody, secondUrlRequest.httpBody)
+      XCTAssertEqualURLRequest(firstUrlRequest, secondUrlRequest)
+    }
+  }
+  
+  func test_XCTAssertEqualURLRequest_hasExpectedFailures() {
```

Comment:
Do you prefer snake case over a camel case?

Reply:
Hmm, this naming convention is something I got from the Essential Developer course. I like it since it makes it very clear what is being tested and what is being expected from the test without having to go into the test itself.

Reply:
I don't mind about the naming at all. Just the case type.
```swift
func test_XCTAssertEqualURLRequest_hasExpectedFailures()
vs 
func testXCTAssertEqualURLRequestHasExpectedFailures()
```
Again, don't have strong feelings. Maybe we should vote about it 😄

Reply:
Sorry, I meant just what you said:

test_`System under test`_`Expected behaviour` the underline makes it easier to read in my opinion 🤷‍♂️ , maybe I just got used to it by now. We could vote on it on our next pod meeting 😄

Reply:
Maybe it's just me that I'm not used to this snake case. But I get your point. It is easier to read. And for tests, it's ok I think 👌 We can still do a quick vote on this at the next pod meeting 😄